contracts/contracts/lib/CrossMsgHelper.sol

```diff
@@ -182,20 +181,19 @@ library CrossMsgHelper {
         }
 
         address recipient = crossMsg.to.rawAddress.extractEvmAddress().normalize();
-        if (crossMsg.kind == IpcMsgKind.Transfer) {
+        // If the cross msg kind is Result, create result message should have handled the value correctly.
+        // If the execution is ok, value should be 0, else one should perform refund.
+        if (crossMsg.kind == IpcMsgKind.Transfer || crossMsg.kind == IpcMsgKind.Result) {
```

Comment:
This is not correct; the version in main is more correct. For a Result message, you will want to perform a call as this returns control back to the caller. Rationale:
- if it's an account, there will be no code to invoke, so this will be have like a bare transfer
- but if the original caller was a contract, you will want to give it control so it can handle the result